### ISSUE TITLE
VDXR incorrectly assumes swapchains are updated every frame; shows out of order.

### ISSUE THREAD

**Issue by fredemmott** [mbucchia/VirtualDesktop-OpenXR#47]
**ANY ISSUE FILED WITHOUT THE INFORMATION REQUIRED BELOW WILL BE CLOSED WITHOUT BEING LOOKED AT. NO EXCEPTIONS.**

**Describe the bug**

Some quad layers (e.g. OpenKneeboard) only update their swap chain when the contents has changed; this is permitted by the spec, but not handled correctly by VDXR. This leads to things going backwards, or visual artifacts.

[OpenXR.log](https://github.com/user-attachments/files/15778679/OpenXR.log)

**Environment**

- OpenKneeboard v1.8.3 or master
- VirtualDesktop 1.32.4
- `hello_xr -G D3D11` from OpenXR SDK v1.0.30

**To Reproduce**

The most visible way:

1. Launch OpenKneeboard
2. Settings -> VR
3. Scroll down; turn off 'Zoom when looking at kneeboard'
4. Scroll down; reduce opacity / 'when not looking at kneeboard' to say 20%
5. launch `hello_xr -G D3D11`
6. Look directly at kneeboard (if needed, bind recentering key in openkneeboard, or click recenter button in VR settings)
7. notice flickering for several frames; if you turn on the frame counter, in OpenKneeboard advaned settings, you can see it go backwards sometimes

This can be resolved by turning on Settings -> Advanced -> Compatibility Quirks -> OpenXR: Always update swapchains

**Expected behavior**

In terms of repro steps for very visible behavior, opacity should reliably change when looking at layer or not

In terms of VDXR behavior, VDXR should handle layers that re-use the same swapchain image for multiple frames.

--- 

Original report in OpenKneeboard discord: https://discord.com/channels/807359994856996915/1249705742785843252

**Comment by mbucchia** [comment]
Thanks Fred, there was a big rework of swapchains between the last 2 versions, something probably broke there in spite of my testing.

**Comment by fredemmott** [comment]
No problem - is this something that is or should be in the conformance tests?

**Comment by mbucchia** [comment]
I appreciate the clear repro steps. I've been looking at this for the past 2 hours and I am clueless.

I checked everything and it appears that VDXR behavior is correct, though I am able to reproduce your issue. Then I toggled VDXR to Link mode (so it uses Meta's LibOVR instead of Virtual Desktop's), and well, I cannot repro the issue there. I believe the issue might be in the Virtual Desktop compositor.
